File: backend/Apprilium.TripPlanner/Apprilium.TripPlanner.Domain/Common/PagedResult.cs
namespace Apprilium.TripPlanner.Domain.Common
{
    public class PagedResult<T> where T : class
    {
        public PagedResult()
        {
        }
        public PagedResult(IEnumerable<T> results, int? pageSize, int totalRows)
        {
            PageSize = pageSize;
            Results = results;
            TotalRows = totalRows;
            PageCount = pageSize != 0 && pageSize != null ? (int)Math.Ceiling((decimal)totalRows / (int)pageSize) : 1;

        }
        public int TotalRows { get; set; }
        public int PageCount { get; set; }
        public int? PageSize { get; set; }
        public IEnumerable<T> Results { get; set; } = new List<T>();
    }
}

File: frentend/Trip_States_Management/packages/common/src/models/PagedResult.ts


 export class PagedResult<T extends object | null> {
    public TotalRows:number=0;
    public PageCount:number=0;
    public  PageSize: number |undefined;
    public  Results :Array<T> |undefined;

    public constructor(result:Array<T>, pagesSize: number| undefined, totalRows:number ){
             this.TotalRows = totalRows;
             this.PageSize = pagesSize;
             this.Results = result;
             this.PageCount = pagesSize != 0 && pagesSize != undefined ? Math.ceil(totalRows/pagesSize):1;
    }

  }